5372 backup level iam operations

amitchauhan1007 opened the pull request.

amitchauhan1007 committed:
updating .gitignore file

amitchauhan1007 committed:
removing stale entries from gitignore

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
added code for backup-level IAM operations

amitchauhan1007 committed:
added code for backup-level IAM operations

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 commented:
seems like some issue with the merge. I made changes only in two files "table_admin.h" & "table_admin.cc"

manish-qlogic commented on google/cloud/bigtable/table_admin.h:
```diff
@@ -1534,6 +1611,13 @@ class TableAdmin {
     return instance_name() + "/clusters/" + cluster_id;
   }
 
+  /// Return the fully qualified name of a Cluster.
```
cluster?

I think it should be `backupName`.

manish-qlogic commented on google/cloud/bigtable/table_admin.h:
```diff
@@ -1389,7 +1389,29 @@ class TableAdmin {
+   * and using different copies in each thread.
+   *
+   * @par Example
+   * @snippet table_admin_iam_policy_snippets.cc get iam policy
```
this `get iam policy` snippet is not for backup, so either we write a snippet for `get iam policy bckup` or at least create a todo.

manish-qlogic commented on google/cloud/bigtable/table_admin.h:
```diff
@@ -1443,6 +1465,36 @@ class TableAdmin {
+   * and using different copies in each thread.
+   *
+   * @par Example
+   * @snippet table_admin_iam_policy_snippets.cc set iam policy
```
same goes with this as well.

manish-qlogic commented on google/cloud/bigtable/table_admin.cc:
```diff
@@ -922,6 +922,31 @@ StatusOr<google::iam::v1::Policy> TableAdmin::GetIamPolicy(
   return proto;
 }
 
+StatusOr<google::iam::v1::Policy> TableAdmin::GetIamPolicy(
```
we should write unit tests in `table_admin_test.cc`.

we should also write integration tests in `admin_iam_policy_integration_test.cc`, it can be a todo for now and can go in separate PR.

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
Updated googleapi version for implementing backup level IAM operations

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
updating TestIamPermissions and adding TestIamPermissionsiForBackup

devjgm committed:
fix: inspect returned grpc::Status (#5558)

In some environments `grpc::Status` is marked as "must_use_result", and
these test lines would cause a breakage due to `-Wunused-result` if we
didn't do something with the `grpc::Status` that's returned from these
`LogWrapper()` calls.

coryan committed:
chore(storage): prepare GCS+gRPC for google import (#5561)

This makes the code in GCS ready for importing into the google codebase.

amitchauhan1007 committed:
updating file format for function TestIamPermissions

devbww committed:
feat(spanner): Point-In-Time Recovery (lite) throttled DB update (#5562)

Update the "com_google_googleapis" package to a recent snapshot.
This includes the `UpdateDatabaseDdlMetadata.throttled` field, and
requires adding `table_reference.proto` to the cmake lists.

The `UpdateDatabaseDdlMetadata.throttled` field indicates that the
`UpdateDatabase()` operation is throttled, e.g., due to resource
constraints. When resources become available the operation will
resume and the field will be false again.

This only updates an integration test. No samples are required for
this field.

coryan committed:
fix(bigtable): missing functions to change policies (#5565)

devbww committed:
chore: december release notes and ABI update (#5566)

scotthart committed:
refactor(generator): remove generation of ConnectionOptions Client constructor (#5567)

devbww committed:
chore: bump version numbers after v1.21 release (#5568)

dopiera committed:
feat: asynchronous waiting for connections (#5569)

This PR implements support for using
`grpc::Channel::NotifyOnStateChange`. This is a prerequisite to avoiding
periodic latency spikes when channels are automatically closed (#5550).

devjgm committed:
chore: fix test name to match file (#5572)

devjgm committed:
doc(api-support): document API support guarantees (#5564)

Fixes: #3847

coryan committed:
fix: get the code to compile with MSVC 2017 (#5574)

The code compiles and the unit tests run with MSVC 2017, specifically,
with:

```
cl.exe -?
Microsoft (R) C/C++ Optimizing Compiler Version 19.16.27044 for x64
```

I did not run the integration tests. Note that I only tested in 64-bit
mode (x64), I did not try a 32-bit build.

coryan committed:
ci: support configuring MSVC version in CI scripts (#5577)

shaffeeullah committed:
fix(storage): allow endings without \r\n in emulator multipart (#5578)

amitchauhan1007 committed:
added code to handle additional_bindings and also unit test code for Get,Set and Test IAM policy

scotthart committed:
feature(generator): add doxygen permalinks for proto message types (#5570)

* feature(generator): add doxygen permalinks for proto message types

amitchauhan1007 committed:
incorporating review comments

amitchauhan1007 committed:
incorporating review comments

amitchauhan1007 committed:
Merge branch 'master' of github.com:googleapis/google-cloud-cpp

amitchauhan1007 committed:
resolving rebase conflicts

amitchauhan1007 committed:
resolving rebase conflicts
